site/src/errors.js

```diff
@@ -20,21 +23,95 @@ export class HTTPError extends Error {
   }
 
   /**
-   * @param {Error & { status?: number }} error
+   * @param {Error & {status?: number;code?: string;}} err
+   * @param {{sentry: Toucan}} ctx
    */
-  static respond(error) {
+  static respond(err, { sentry }) {
+    let error = {
+      code: err.code,
+      message: err.message,
+    }
+    let status = err.status || 500
+
+    switch (err.code) {
+      case ErrorUserNotFound.CODE:
+      case ErrorTokenNotFound.CODE:
+        break
+
+      // Magic SDK errors
+      case MagicErrors.TokenExpired:
+        status = 401
+        error.message = 'API Key has expired.'
+        break
+      case MagicErrors.ExpectedBearerString:
+        status = 401
+        error.message =
+          'API Key is missing, make sure the `Authorization` header has a value in the following format `Bearer {api key}`.'
+        break
+      case MagicErrors.MalformedTokenError:
+        status = 401
+        error.message = 'API Key is malformed or failed to parse.'
+        break
+      case MagicErrors.TokenCannotBeUsedYet:
+      case MagicErrors.IncorrectSignerAddress:
+      case MagicErrors.FailedRecoveryProof:
+      case MagicErrors.ApiKeyMissing:
+        status = 401
+        error.code = 'AUTH_ERROR'
+        error.message = 'Authentication failed.'
+        sentry.captureException(err)
+        break
+      case MagicErrors.ServiceError:
+        status = 500
+        error.code = 'SERVER_ERROR'
+        sentry.captureException(err)
+        break
+      default:
+        // catch all server errors
+        if (status >= 500) {
+          error = {
+            code: err.name,
+            message: err.message,
+          }
+          sentry.captureException(err)
+        } else {
+          // Custom HTTPError
+          error = {
+            code: 'HTTP_ERROR',
+            message: err.message,
+          }
+        }
+        break
+    }
+
     return new JSONResponse(
       {
         ok: false,
-        error: {
-          name: error.name,
-          message: error.message,
-        },
+        error,
       },
       {
-        statusText: error.message,
-        status: error.status || 500,
+        status,
       }
     )
   }
 }
+
+export class ErrorUserNotFound extends Error {
```

Comment:
with TS support i like it this way because if im trying to find the proper error i just start typing `Error...` and i get a list of all error available but i can change this